### DISCUSSION TITLE
Using new supabase features such as `previews` and `migrations`

### DISCUSSION THREAD

**Discussion by ydennisy** [supabase/supabase#17303] [General]
Hi Team,

As always great work on shipping very cool looking features!

Although I think they look great from a supabase perspective, I do not want to be fully wedded to any technology or provider. So when I saw the new features I was concerned that if I use these I will be locked in for example, if I switch my workflow from `alembic` migrations to your flavour will I be locked in?

It would be great to hear from someone on your team about how to be able to leverage the new features without lock in.

Thanks!
D

**Comment by kiwicopple** [comment]
hey @ydennisy - 

> how to be able to leverage the new features without lock in.

previews work with any postgres migration tool. (for example, you can use them with Alembic). 

The opposite is also true - our migrations work with any Postgres database - they are just SQL.

There is nothing proprietary about the way we do Database Previews - we simply expose another URL for the preview database and the SQL migrations are run against that database. 

In it's current state our implementation relies on GitHub integration. We have built a webhooks listener and a migrations runner for the integration to work. This part of the code is tightly coupled with some Supabase business logic (for billing. and it's configured with our GitHub API key). We'll open source this Action it in the future so you can setup webhooks integration on self-hosted Supabase instances as well. We're still in the "testing" phase of Preview Environments, so things are changing rapidly and subject to breaking changes - for now it's easier to iterate with a few partners until we reach an alpha release.

**Reply by ydennisy** [reply]
hey @kiwicopple I appreciate very much your reply and congratulations on the good work!

I have no plans to move away from Supabase or self host, but when using PG you want to know you can if we ever hit a reason. So this is very important.

Thanks again!
